### DISCUSSION TITLE
Machine set not working after upgrade to 4.11

### DISCUSSION THREAD

**Discussion by dkoci** [okd-project/okd#1445] [General]
We have some problems with machines creation and deletion after okd upgrade to 4.11 (vmware). In case of creation there is no error but machines are not created.
In case of scaling down machines stuck in deleting status.
Anybody has the same issue? This was working fine on previous okd versions.

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

**Reply by dkoci** [reply]
@vrutkovs Must-gather archive is available at https://www.transfernow.net/dl/20221227VqcEsb26

**Comment by dkoci** [comment]
Anyone has the same issue? We really need to add aditional worker node, but it is impossible to do that with machine-set in OKD 4.11.
![image](https://user-images.githubusercontent.com/5565897/209705043-b8965b85-b4a5-4680-9987-570088386709.png)

**Reply by nate-duke** [reply]
We've added quite a few nodes via MachineSet just since the last update (06 December 2022) but we've been running 4.11 since early August. There should be something telling in your logs. I've not tried deciphering must-gather contents before but when we've had provisioning issues before there were always signs either in the openshift-machine-api pods or openshift-vsphere-infra pods (we use on-prem, vsphere IPI)  
![image](https://user-images.githubusercontent.com/40432118/209711160-789dc77e-ac66-4d3f-b305-386387a86f6e.png)

**Reply by dkoci** [reply]
The only thing which looks like an issue is in machine-api-operator pod. There are no errors in other pods.

<img width="921" alt="image" src="https://user-images.githubusercontent.com/5565897/209817833-00faa751-8510-4375-8981-a19f61e3889c.png">

**Reply by nate-duke** [reply]
flip that container target over to whatever the proxy container (machine-api-operator probably) is trying to connect to in that pod. *that* should have whatever errors are preventing it from running (and answering on :8080)

![image](https://user-images.githubusercontent.com/40432118/209818757-c1d7f3c9-1dc3-4ef4-b419-e0cc32500735.png)

**Reply by dkoci** [reply]
machine-api-operator logs:

I1228 18:08:17.143679 1 status.go:99] Syncing status: available
I1228 18:08:17.331065 1 request.go:533] Waited for 187.215251ms due to client-side throttling, not priority and fairness, request: PUT:https://172.30.0.1:443/apis/config.openshift.io/v1/clusteroperators/machine-api/status

**Reply by nate-duke** [reply]
yeah, stuff like that is pretty normal. Probably a good idea to look around at the other pods in that namespace and maybe the `openshift-vsphere-infra` namespace.

**Reply by dkoci** [reply]
Finally I found it. Our infrastructure team changed vmware url. I replaced it in machine-set template, but vsphere-cloud-credentials kept the old one. I had to change <vmware-url>.username and <vmware-url>.password in vsphere-cloud-credentials. It is working now.
@nate-duke thank you for your tips.

**Reply by nate-duke** [reply]
Hey, no problem. Glad you found it. If you have access to redhat's KCS there's [an article](https://access.redhat.com/solutions/4618011) there that may help you with other vsphere credential changes post-install. There was a github issue or maybe discussion at some point here on that topic but it was unsupported at the time. That KCS article outlines the process though so I guess it's marginally supported, at least in OCP at this point.
